Pod/Classes/ios/NYTPhotosViewController.m

```diff
@@ -239,8 +240,32 @@ - (void)actionButtonTapped:(id)sender {
                 [self.delegate photosViewController:self actionCompletedWithActivityType:activityType];
             }
         };
-        
-        [self presentViewController:activityViewController animated:YES completion:nil];
+
+        [self displayActivityViewController:activityViewController animated:YES];
+    }
+}
+
+- (void)displayActivityViewController:(UIActivityViewController *)controller animated:(BOOL)animated {
+
+    if (UI_USER_INTERFACE_IDIOM() == UIUserInterfaceIdiomPhone) {
+        // iPhone
+        [self presentViewController:controller animated:animated completion:nil];
+    }
+    else {
+        // iPad
+        if ([controller respondsToSelector:@selector(popoverPresentationController)]) {
+
+            // iOS8 and above
+            controller.popoverPresentationController.barButtonItem = self.rightBarButtonItem;
+            [self presentViewController:controller animated:animated completion:nil];
+        }
+        else {
+            // iOS 7
+            _activityPopoverController = [[UIPopoverController alloc] initWithContentViewController:controller];
```

Comment:
@heikkihautala is there a particular reason you're using the instance variable here rather than using `self.activityPopoverController` as a property?